### DISCUSSION TITLE
Hanging the whole instance while running the script in the usage section

### DISCUSSION THREAD

**Discussion by luk0y** [gotd/td#867] [Q&A] [answered]
**What happened?**

I am trying to run the example code but have had no luck. It always keeps hanging the AWS instance

**Version and System configurations**

Go Version: go version go1.19 linux/amd64
System Configurations : AWS Instance 2 GB RAM, 1 vCPU, 60 GB SSD
Operating System: Ubuntu 20.04

**Steps to reproduce**

1) Installing the 1.19 version of go through go.dev. followed the installation procedure till setting the paths in .profile

2) Saved the code from the Usage section https://github.com/gotd/td#usage by naming it as hello.go

```
package main

import (
	"context"

	"github.com/gotd/td/telegram"
)

func main() {
	// https://core.telegram.org/api/obtaining_api_id
	client := telegram.NewClient(appID, appHash, telegram.Options{})
	if err := client.Run(context.Background(), func(ctx context.Context) error {
		// It is only valid to use client while this function is not returned
		// and ctx is not cancelled.
		api := client.API()

		// Now you can invoke MTProto RPC requests by calling the API.
		// ...

		// Return to close client connection and free up resources.
		return nil
	}); err != nil {
		panic(err)
	}
	// Client is closed.
}
```
3) Assigned values to variables appID(Int), appHash(String) in the file hello.go

4) `go mod init hello.go `

5) `go mod tidy`

6) Ran the code hello.go using `go run hello.go` and I can't see anything on the screen. It just hangs

When I ran the code in tmux and tried checking from outside of tmux to know if the hang is because of too much usage of CPU/memory using the command top. It's shown that the command "compile" is taking too much CPU power and memory

![Screenshot from 2022-08-29 09-56-19](https://user-images.githubusercontent.com/45989946/187125139-c6607f7e-741b-48c9-a168-0f3c2fc743eb.png)

If I keep running it like this for 5-10 mins, I can't be able to access my instance again. Tried the same procedure in multiple instances but the issue was not resolved

**Comment by giautm** [comment, marked as answer]
Can you try to compile the source first? Use `go build` to compile it to a binary file then execute.

**Reply by luk0y** [reply]
@giautm It's still the same. My instance got stuck again due to the high usage of CPU

![Screenshot from 2022-08-29 11-59-23](https://user-images.githubusercontent.com/45989946/187137169-1aea1f2e-fdaf-4847-8572-e7a888ca6466.png)

**Reply by giautm** [reply]
Can you try compile the source in your local machine and upload the binary to your server?

**Reply by giautm** [reply]
You need to add some compile flag to ensure Golang build correct Arch for linux, please Google it

**Reply by luk0y** [reply]
Thank you so much. The executable is working fine on the server, but I don't understand the issue while compiling the same on the server.

**Reply by giautm** [reply]
The AWS instance may take long time to compile your source, it doesn't have enought performance to do that.
